lib/semantics/check-do.cc

```diff
@@ -43,19 +43,23 @@ static const parser::Name *MaybeGetConstructName(
           .statement.v);
 }
 
-// Return the (possibly null) name of the statement
-template<typename A> static const parser::Name *MaybeGetStmtName(const A &a) {
-  return common::GetPtrFromOptional(std::get<0>(a.t));
-}
-
 // 11.1.7.5 - enforce semantics constraints on a DO CONCURRENT loop body
 class DoConcurrentBodyEnforce {
 public:
   DoConcurrentBodyEnforce(SemanticsContext &context) : context_{context} {}
   std::set<parser::Label> labels() { return labels_; }
   std::set<SourceName> names() { return names_; }
   template<typename T> bool Pre(const T &) { return true; }
-  template<typename T> void Post(const T &) {}
+  template<typename T> void Post(const T &node) {
+    if (const std::optional<parser::MessageFixedText> message{
+            GetImageControlStmtMsg(node)}) {  // C1137
```

Comment:
Can this constraint have a brief description as the ones below do?

Reply:
Certainly!  I'll flesh out all of these cryptic constraint notations.